uk.co.compucorp.civicrm.hrcomments/tests/phpunit/CRM/HRComments/BAO/CommentTest.php

```diff
@@ -1,53 +1,91 @@
 
   public function setUp() {
-    parent::setUp();
+    CRM_Core_DAO::executeQuery("SET foreign_key_checks = 0;");
+  }
+
+  /**
+   * @expectedException CRM_HRComments_Exception_InvalidCommentException
+   * @expectedExceptionMessage Comment should have associated entity ID
+   */
+  public function testValidateCommentWhenEntityIdIsNotPresent() {
+    Comment::validateParams([
+      'entity_name' => 'LeaveRequest',
+      'text' => 'This is a sample comment',
+      'contact_id' => 1,
+    ]);
   }
 
-  public function tearDown() {
-    parent::tearDown();
+  /**
+   * @expectedException CRM_HRComments_Exception_InvalidCommentException
+   * @expectedExceptionMessage Comment should have associated entity name
+   */
+  public function testValidateCommentWhenEntityNameIsNotPresent() {
+    Comment::validateParams([
+      'entity_id' => 1,
+      'text' => 'This is a sample comment',
+      'contact_id' => 1,
+    ]);
   }
 
   /**
-   * Example: Test that a version is returned.
+   * @expectedException CRM_HRComments_Exception_InvalidCommentException
+   * @expectedExceptionMessage Comment should have text
    */
-  public function testWellFormedVersion() {
-    $this->assertRegExp('/^([0-9\.]|alpha|beta)*$/', \CRM_Utils_System::version());
+  public function testValidateCommentWhenTextIsNotPresent() {
+    Comment::validateParams([
+      'entity_id' => 1,
+      'entity_name' => 'LeaveRequest',
+      'contact_id' => 1,
+    ]);
   }
 
   /**
-   * Example: Test that we're using a fake CMS.
+   * @expectedException CRM_HRComments_Exception_InvalidCommentException
+   * @expectedExceptionMessage Contact who made the comment should not be empty
    */
-  public function testWellFormedUF() {
-    $this->assertEquals('UnitTests', CIVICRM_UF);
+  public function testValidateCommentWhenContactIdIsNotPresent() {
+    Comment::validateParams([
+      'entity_id' => 1,
+      'entity_name' => 'LeaveRequest',
+      'text' => 'This is a sample comment',
+    ]);
   }
 
+  /**
+   * @expectedException CRM_HRComments_Exception_InvalidCommentException
+   * @expectedExceptionMessage Contact who made the comment should not be empty
+   */
+  public function testValidateParamsIsCalledOnCreate() {
+    Comment::create([
+      'entity_id' => 1,
+      'entity_name' => 'LeaveRequest',
+      'text' => 'This is a sample comment',
+    ]);
+  }
+
+  public function testCreatedDateIsInsertedCorrectly() {
+    $comment = Comment::create([
+      'entity_id' => 1,
+      'entity_name' => 'LeaveRequest',
+      'text' => 'This is a sample comment',
+      'contact_id' => 1
+    ]);
+
+    //the current timestamp is gotten here and asserted
+    //that it is within 10 seconds delta compared to the created_date of the comment entity
+    $createdDateTimestamp = new DateTime($comment->created_at);
+    $timestampNow = new DateTime('now');
+
+    $this->assertEquals($timestampNow, $createdDateTimestamp, '', 10);
+  }
 }
```

Comment:
I would suggest two more tests to be added here:
- Test that `created_at` will be the current date even when a specific date is passed to `Comment::create()`
- Test that the `created_at` value cannot be changed when updating an existing Comment